File: forge-cli/pkg/profile/languages/python/templates/journey_test.py
import pytest
import subprocess
import re

pytestmark = pytest.mark.feature


def test_journey_smoke(tmp_path):
    """Smoke test for Journey: happy path end-to-end."""
    # VERIFY: setup project structure in tmp_path

    # Step N: <action>
    # step_N = subprocess.run(["<binary>", "<args>"], capture_output=True, text=True)
    # assert step_N.returncode == 0, f"Step N failed: {step_N.stderr}"
    # assert re.search(r"<pattern from Fact Table>", step_N.stdout)
    pass

File: forge-cli/pkg/profile/languages/go/templates/journey-test.go
//go:build feature

package journey_test

import (
	"os/exec"
	"testing"

	"github.com/stretchr/testify/assert"
)

// TestJourney_Smoke verifies the happy path end-to-end for a Journey.
// Each step output is validated against the Contract "success" Outcome.
func TestJourney_Smoke(t *testing.T) {
	dir := t.TempDir()
	_ = dir // VERIFY: setup project structure as needed

	// Step N: <action>
	// stepCmd := exec.Command("<binary>", "<args>")
	// stepOut, err := stepCmd.CombinedOutput()
	// if err != nil {
	//     t.Fatalf("Step N failed: %s\nOutput: %s", err, stepOut)
	// }
	// assert.Regexp(t, "<pattern from Fact Table>", string(stepOut))

	_ = exec.Command // CLI tests
	_ = assert.Equal // assertions
}
